examples/text_to_image/train_text_to_image_sdxl.py

```diff
@@ -540,7 +541,7 @@ def compute_vae_encodings(batch, vae):
     with torch.no_grad():
         model_input = vae.encode(pixel_values).latent_dist.sample()
     model_input = model_input * vae.config.scaling_factor
-    return {"model_input": model_input.cpu()}
+    return {"model_input": accelerator.gather(model_input)}
```

Comment:
Why do we need this?

Reply:
By using the accelerator, the communication time can be reduced

Reply:
I think the reason may caused by the pixel_values, as it is in vae.device (accelerator). Therefore, by changing the code, the accelerator can distribute and reduce the time cost.

Reply:
But isn't an all-gather a more expensive op?

Reply:
In fact, I tested three different approaches, the first one is accelerator.gather(model_input), the average FPS is 29.13 with training duration 530; the second one is model_input.to(accelerator.device), the average FPS is 27.41 with training duration 544; the last one is the original model_input.cpu(), the average FPS is 28.56 with training duration 537. Overall, with same hardware, the FPS will increase a little with accelerator.gather. I tested multiple times with accelerator.gather and model_input.cpu(), the average FPS in accelerator.gather is larger than model_input.cpu().

Reply:
Hmm, thanks! Since the performance improvement seems to be minor, do you think it makes sense to not change this?

Reply:
Make sense, I will change it back to the .cpu

Reply:
But I welcome you to also add a note on your findings on `accelerate.gather()` so that users are aware. I think that'd still be quite valuable.